git_machete/client.py

```diff
@@ -1656,6 +1659,60 @@ def flush_caches(self) -> None:
         self.__branch_defs_by_sha_in_reflog = None
         self.__git.flush_caches()
 
+    def checkout_github_pr(self, pr_no: int) -> None:
+        org: str
+        repo: str
+        remote: str
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        debug('checkout_github_pr()', f'organization is {org}, repository is {repo}')
+
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
+        pr: GitHubPullRequest = find_pr_by_number(pr_no, all_prs)
+
+        if not pr:
+            raise MacheteException(f"PR number {pr_no} is not found in repository {repo}")
+        debug('checkout_github_pr()', f'found {pr}')
+
+        print(f"Fetching {remote}...")
+        self.__git.fetch_remote(remote)
+
+        self.__cli_opts.opt_yes = True  # TODO (#161): pass only needed options to methods
+        path: List[str] = self.__get_path_from_pr_chain(pr.head, all_prs)
+        reversed_path: List[str] = path[::-1]  # need to add from root downwards
+        for index, branch in enumerate(reversed_path):
+            if branch not in self.managed_branches:
+                if index == 0:
+                    self.__cli_opts.opt_as_root = True
+                    self.add(branch)
+                    self.__cli_opts.opt_as_root = False
+                else:
+                    self.__cli_opts.opt_onto = reversed_path[index - 1]
+                    self.add(branch)
```

Comment:
Whooops I've seen a bug here: clone repo https://github.com/VirtusLab/akka-serialization-helper/ , add just `main` to .git/machete and then run:

```
git branch -r -D origin/feachure/state-dump
git machete github checkout-pr 95
```

You'll probably see:

```
$ git machete github checkout-pr 95
Fetching origin...
From github.com:VirtusLab/akka-serialization-helper
 * [new branch]      feachure/state-dump -> origin/feachure/state-dump
A local branch feachure/state-dump does not exist. Creating out of main
Switched to a new branch 'feachure/state-dump'
Added branch feachure/state-dump onto main
Annotating feachure/state-dump as PR #95 (MarconZet)
Pull request #95 checked out at local branch feachure/state-dump
```

Reply:
`feachure/state-dump` will be checked out of **`main`** and NOT of `origin/feachure/state-dump`... that's probably because `origin/feachure/state-dump` only appears once `fetch_remote` is executed.

Reply:
`__remote_branches_cached` should be updated after a fetch happens... see how it's handled e.g. in `def traverse`.

Reply:
Pls also first a regression test and let it fail, in the spirit of [TDD](https://pl.wikipedia.org/wiki/Test-driven_development) ‼️